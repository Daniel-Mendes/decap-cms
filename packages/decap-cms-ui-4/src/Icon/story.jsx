import React from 'react';
import styled from '@emotion/styled';
import { withKnobs, select } from '@storybook/addon-knobs';

import Icon, { iconComponents } from './Icon';

const StyledIcon = styled(Icon)`
  color: ${({ theme }) => theme.color.highEmphasis};
`;

export default {
  title: 'Components/Icon',
  decorators: [withKnobs],
};

export const _Icon = () => {
  return (
    <StyledIcon
      size={select('size', { xs: 'xs', sm: 'sm', md: null, lg: 'lg', xl: 'xl' }, null)}
      name={select(
        'name',
        {
          default: null,
          ...Object.keys(iconComponents).reduce((acc, key) => ({ ...acc, [key]: key }), {}),
        },
        null,
      )}
    />
  );
};
