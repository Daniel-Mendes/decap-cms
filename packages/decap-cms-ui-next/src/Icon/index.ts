export { default, iconComponents } from './Icon';
export type { IconName } from './Icon';
