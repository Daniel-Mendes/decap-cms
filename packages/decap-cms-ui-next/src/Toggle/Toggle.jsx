import React from 'react';
import styled from '@emotion/styled';
import { css } from '@emotion/react';
import color from 'color';

import { Button } from '../Buttons';

const StyledButton = styled(Button)`
  ${({ theme, active }) => css`
    color: ${active ? theme.color.success['900'] : theme.color.highEmphasis};
    background-color: ${active
      ? color(theme.color.success['900']).alpha(0.1).string()
      : 'transparent'} !important;
  `}

  &:hover,
  &:focus {
    color: ${({ theme, active }) =>
      active ? theme.color.success['900'] : theme.color.highEmphasis};
    background-color: ${({ theme, active }) =>
      active
        ? color(theme.color.success['900']).alpha(0.2).string()
        : color(theme.color.highEmphasis).alpha(0.05).string()} !important;
  }
`;

function Toggle({ icon, hasMenu, pressed, onPressedChange, disabled, children }) {
  return (
    <StyledButton
      icon={icon}
      hasMenu={hasMenu}
      active={pressed}
      disabled={disabled}
      onClick={event => onPressedChange(event)}
    >
      {children}
    </StyledButton>
  );
}

export default Toggle;
