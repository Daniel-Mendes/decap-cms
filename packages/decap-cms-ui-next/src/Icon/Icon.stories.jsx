import React from 'react';

import Icon, { iconComponents } from './Icon';

export default {
  title: 'Components/Icon',
  component: Icon,
  argTypes: {
    name: {
      control: 'select',
      options: {
        default: null,
        ...Object.keys(iconComponents).reduce((acc, key) => ({ ...acc, [key]: key }), {}),
      },
      table: {
        defaultValue: { summary: 'null' },
      },
    },
    size: {
      control: 'select',
      options: ['xs', 'sm', 'md', 'lg', 'xl'],
      mapping: {
        md: null,
      },
      table: {
        defaultValue: { summary: 'md' },
      },
    },
  },
  args: {
    name: null,
    size: 'md',
  },
};

export function _Icon(args) {
  return <Icon {...args} />;
}
