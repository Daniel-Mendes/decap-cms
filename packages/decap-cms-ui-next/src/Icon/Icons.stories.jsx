import React from 'react';

import Icon, { iconComponents } from './Icon';

export default {
  title: 'Foundations/Icons Library',
  component: Icon,
  parameters: {
    options: {
      showPanel: false,
    },
  },
};

export function IconsLibrary() {
  return (
    <div
      style={{
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        margin: '0 10rem',
      }}
    >
      <h1>Icons Library</h1>

      <div
        style={{
          display: 'flex',
          flexWrap: 'wrap',
          flexDirection: 'row',
          justifyContent: 'center',
          alignItems: 'center',
        }}
      >
        {Object.keys(iconComponents).map(name => (
          <div
            key={name}
            style={{
              flex: '1 1 25%',
              display: 'flex',
              flexDirection: 'row',
              alignItems: 'flex-end',
              gap: '1rem',
            }}
          >
            <Icon name={name} size="md" />
            <span style={{ marginTop: 4 }}>{name}</span>
          </div>
        ))}
      </div>
    </div>
  );
}
