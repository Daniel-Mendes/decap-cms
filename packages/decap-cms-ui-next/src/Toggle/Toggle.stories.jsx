import React from 'react';
import { action } from '@storybook/addon-actions';
import { useArgs } from '@storybook/preview-api';

import Toggle from '.';
import { iconComponents } from '../Icon';

export default {
  title: 'Components/Toggle',
  component: Toggle,
  argTypes: {
    pressed: {
      control: 'boolean',
      table: {
        defaultValue: { summary: false },
      },
    },
    disabled: {
      control: 'boolean',
      table: {
        defaultValue: { summary: false },
      },
    },
    icon: {
      control: 'select',
      options: {
        default: null,
        ...Object.keys(iconComponents).reduce((acc, key) => ({ ...acc, [key]: key }), {}),
      },
      table: {
        defaultValue: { summary: 'null' },
      },
    },
  },
  args: {
    icon: 'bold',
    hasMenu: false,
    pressed: false,
    onPressedChange: action('onPressedChange'),
    disabled: false,
  },
};

export function _Toggle(args) {
  const [{ pressed }, updateArgs] = useArgs();

  function togglePressed() {
    updateArgs({ pressed: !pressed });
  }

  return <Toggle {...args} onPressedChange={togglePressed} />;
}
