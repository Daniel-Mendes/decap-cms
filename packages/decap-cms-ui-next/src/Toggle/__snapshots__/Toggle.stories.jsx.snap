// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Toggle _Toggle smoke-test 1`] = `"<button type="primary" class="eh9hc6f0 css-1xb9di3-StyledButton-StyledButton-StyledButton-StyledButton e9pg3mh2"><svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" class="css-wwhi30-StyledIcon e9pg3mh0" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><path d="M6 4h8a4 4 0 0 1 4 4 4 4 0 0 1-4 4H6z"></path><path d="M6 12h9a4 4 0 0 1 4 4 4 4 0 0 1-4 4H6z"></path></svg></button>"`;
